### PR TITLE
[Serving] Minor fixes

### PR DESCRIPTION
**Pull request by yaronha**
* fix small bugs in serving
* add v3io stream trigger to nuclio
* small enhancments to internal methods

### CODE REVIEW

File: mlrun/__init__.py [hunk 1]
```diff
@@ -61,6 +61,18 @@ def get_version():
...
 
+_data_stores = None
+
+
+def get_data_stores(secrets=None, override=False):
+    """singleton mlrun data store client"""
+
+    global _data_stores
+    if not _data_stores or override:
+        _data_stores = store_manager.set(secrets)
```
Comment (Hedingber) [inline mlrun/__init__.py:72]: `.set()` returning `self`, i.e. a `store_manager` why you're calling it `_data_stores` ?
shouldn't it simply be `_store_manager` ?

File: mlrun/__init__.py [hunk 2]
```diff
@@ -61,6 +61,18 @@ def get_version():
     mlconf.remote_host = mlconf.remote_host or igz_domain
 
 
+_data_stores = None
+
+
+def get_data_stores(secrets=None, override=False):
+    """singleton mlrun data store client"""
```
Comment (Hedingber) [inline mlrun/__init__.py:68]: `store_manager` is already cached and is a singleton, why this function is needed ? 
also setting the secret is not an expensive operation (like HTTPRunDB's connect), I don't see a reason not doing it (what if secrets were changed ?)

File: mlrun/runtimes/function.py [hunk 3]
```diff
@@ -230,6 +232,25 @@ def serving(
 
         return self
 
+    def add_stream_trigger(
```
Comment (Hedingber) [inline mlrun/runtimes/function.py:235]:
```suggestion
    def add_v3io_stream_trigger(
```
There might be other stream triggers

File: mlrun/model.py [hunk 4]
```diff
@@ -25,6 +25,16 @@
...
 
 
+class ResourceSchema:
+    FeatureSet = "fset"
+    FeatureVector = "fvec"
+    Artifact = "store"
+
+    @classmethod
+    def is_resource(cls, kind):
+        return kind in [cls.Artifact, cls.FeatureSet, cls.FeatureVector]
```
Comment (Hedingber) [inline mlrun/model.py:35]: What is this ?

**Review by Hedingber — CHANGES_REQUESTED** [review]

File: mlrun/__init__.py (hunk 1 above)
Comment (yaronha) [reply mlrun/__init__.py:72]: ok, missed that, i see most of the code is not using the singleton, so i will update it as well (same as removing the .connect() for the db)

Comment (yaronha) [reply mlrun/__init__.py:72]: the issue is that `secrets ` and `db` are not passed to it, so i will add `get(..)` method to the current singleton

Comment (yaronha) [reply mlrun/__init__.py]: fixed, use the cached store_manager everywhere

File: mlrun/__init__.py (hunk 2 above)
Comment (yaronha) [reply mlrun/__init__.py]: fixed

File: mlrun/runtimes/function.py (hunk 3 above)
Comment (yaronha) [reply mlrun/runtimes/function.py]: fixed

File: mlrun/model.py (hunk 4 above)
Comment (yaronha) [reply mlrun/model.py:35]: its for the next PR, define the special prefixes (url schema) for feature store and artifact store data urls

**Pull request merged**